romancal/patch_match/patch_match.py

```diff
@@ -0,0 +1,291 @@
+    """
+
+    if isinstance(image_corners, wcs.WCS):
+        iwcs = image_corners
+        # Now must find size of correspinding image, with three possible
+        # sources of that information.
+        if (
+            (not hasattr(iwcs, "bounding_box") or iwcs.bounding_box is None)
+            and (not hasattr(iwcs, "pixel_shape") or iwcs.pixel_shape is None)
+            and image_shape is None
+        ):
+            raise ValueError(
+                "Use of a wcs object requires at least one of the bounding_box"
+                " or pixel_shape attributes be set to the image shape or that the"
+                "image_shape argument be set"
+            )
+        if image_shape is not None:
+            pass
+        else:
+            if hasattr(iwcs, "bounding_box") and iwcs.bounding_box is not None:
+                # Presumes that the bounding_box matches image array boundaries
+                bbintervals = iwcs.bounding_box.intervals
+                # This compensates for the half pixel adjustment in the general code.
+                image_shape = (bbintervals[0].upper + 0.5, bbintervals[1].upper + 0.5)
+            elif hasattr(iwcs, "pixel_shape") and iwcs.pixel_shape is not None:
+                image_shape = iwcs.pixel_shape
+        # Compute the image corners ra, dec from the wcs
+        (cxm, cxp), (cym, cyp) = (
+            (-0.5, image_shape[1] - 0.5),
+            (-0.5, image_shape[0] - 0.5),
+        )
+        image_corners = (iwcs(cxp, cyp), iwcs(cxm, cyp), iwcs(cxm, cym), iwcs(cxp, cym))
+    ptab = PATCH_TABLE
+    ra = ptab[:]["ra_center"]
+    dec = ptab[:]["dec_center"]
+    # # Convert all celestial coordinates to cartesion coordinates.
+    vec_centers = np.array(sgv.lonlat_to_vector(ra, dec)).transpose()
+    # # Organize corners into two ra, dec lists
+    # imra = np.array([point[0] for point in image_corners])
+    # imdec = np.array([point[1] for point in image_corners])
+    vec_im_corners = image_coords_to_vec(image_corners)
+    # Approximate center of image by averaging corner vectors
+    im_center = normalize_vector(vec_im_corners.mean(axis=1))
+    # Compute difference vector between image center and patch centers
+    diff = vec_centers - im_center
+    dist = np.sqrt((diff**2).sum(axis=1)) * 180 / np.pi
+    match = np.where(dist < 0.5)
+    ncandidates = len(match[0])
+    # Now see which of these that are close actually overlap the supplied image.
+    # (Is it necessary to check that the corners are in a sensible order?)
+    mra1 = ptab[match]["ra_corn1"]
+    mra2 = ptab[match]["ra_corn2"]
+    mra3 = ptab[match]["ra_corn3"]
+    mra4 = ptab[match]["ra_corn4"]
+    mdec1 = ptab[match]["dec_corn1"]
+    mdec2 = ptab[match]["dec_corn2"]
+    mdec3 = ptab[match]["dec_corn3"]
+    mdec4 = ptab[match]["dec_corn4"]
+    mcenters = vec_centers[match]
+    mra = np.vstack([mra1, mra2, mra3, mra4, mra1])
+    mdec = np.vstack([mdec1, mdec2, mdec3, mdec4, mdec1])
```

Comment:
Are `mra*` and `mdec*` scalars or vectors? Why do you use `vstack` here? `vstack` applied to a list of scalars results in a 2D array. The documentation on `lonlat_to_vector` arguments says:
```
lon, lat : scalars or 1-D arrays
```
It may work with higher dimensionality arrays but this is an undocumented feature.

Reply:
> Did you mean "applied to a list of 1-D arrays"?

No. I did not mean that. I assumed `mra1` is a scalar => `mra` is a 2D array (why do you need a 2D array, assuming `mra1` is a scalar). I used "assuming" and that is why I started my question with _"Are `mra*` and` mdec*` scalars or vectors?"_